File: src/model_v4/entities/SBS_VIEWS.ts
import { Column, Entity, PrimaryColumn } from "typeorm";


@Entity("SBS_VIEWS",{schema:"sbsdb"})
export class SBS_VIEWS {

    @PrimaryColumn("bigint",{ 
        nullable:false,
        primary:true,
        name:"VIEW_INDEX"
        })
    VIEW_INDEX:string;
        

    @Column("varchar",{ 
        nullable:false,
        length:50,
        name:"VIEW_NAME"
        })
    VIEW_NAME:string;
        

    @Column("longtext",{ 
        nullable:false,
        name:"VIEW_SQL"
        })
    VIEW_SQL:string;
        

    @Column("varchar",{ 
        nullable:false,
        length:1,
        name:"VIEW_TYPE"
        })
    VIEW_TYPE:string;
        
}

File: SbsdbServer/sbsdbv4/model/SbsViews.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace hb.SbsdbServer.sbsdbv4.model {
    [Table("SBS_VIEWS")]
    public class SbsViews {
        [Column("VIEW_INDEX", TypeName = "bigint(20)")]
        public long ViewIndex { get; set; }

        [Required]
        [Column("VIEW_NAME", TypeName = "varchar(50)")]
        public string ViewName { get; set; }

        [Required]
        [Column("VIEW_SQL", TypeName = "longtext")]
        public string ViewSql { get; set; }

        [Required]
        [Column("VIEW_TYPE", TypeName = "varchar(1)")]
        public string ViewType { get; set; }
    }
}
